03-rnaseq/pathway-analysis_rnaseq_01_ora.Rmd

```diff
@@ -0,0 +1,385 @@
+}
+
+# This is required to make one of the plots that clusterProfiler will make
+if (!("ggupset" %in% installed.packages())) {
+  # Install this package if it isn't installed yet
+  BiocManager::install("ggupset", update = FALSE)
+}
+
+if (!("msigdbr" %in% installed.packages())) {
+  # Install this package if it isn't installed yet
+  BiocManager::install("msigdbr", update = FALSE)
+}
+
+if (!("org.Hs.eg.db" %in% installed.packages())) {
+  # Install this package if it isn't installed yet
+  BiocManager::install("org.Hs.eg.db", update = FALSE)
+}
+```
+
+Attach the packages we need for this analysis.
+
+```{r}
+# Attach the library
+library(clusterProfiler)
+
+# Package that contains MSigDB gene sets in tidy format
+library(msigdbr)
+
+# Homo sapiens annotation package we'll use for gene identifier conversion
+library(org.Hs.eg.db)
+
+# We will need this so we can use the pipe: %>%
+library(magrittr)
+```
+
+**REVIEW** 
+## Import data
+
+For ORA, we only need a list of genes as our input, so this example can work for any situations where you have gene list and want to know more about what biological pathways it shares genes with.
+
+For this example, we will read in results from a co-expression network analysis that we have already performed.
+Rather than reading from a local file, we will download the results table directly from a URL.
+These results are from a acute bronchiolitis experiment we used for [an example WGCNA analysis](https://alexslemonade.github.io/refinebio-examples/04-advanced-topics/network-analysis_rnaseq_01_wgcna.html) [@Langfelder2008].
+
+The table contains two columns, one with Ensembl gene IDs, and the other with the name of the module they are a part of.
+We will perform ORA on one of the modules we identified in the WGCNA analysis but the rest of the genes will be used as "background genes".
+
+Instead of using this URL below, you can use a file path to a TSV file with your desired gene list.
+First we will assign the URL to its own variable called, `gene_module_url`.
+
+```{r}
+# Define the url to your gene list file
+gene_module_url <- "https://refinebio-examples.s3.us-east-2.amazonaws.com/04-advanced-topics/results/SRP140558_wgcna_gene_to_module.tsv"
+```
+
+Read in the file that has WGCNA gene and module results.
+Here we are using the URL we set up above, but this can be a local file path instead _i.e._ you can replace `gene_module_url` in the code below with a path to file you have on your computer like: `file.path("results", "SRP140558_results_gene_list.tsv")`.
+
+```{r}
+# Read in the contents of your differential expression results file
+# `gene_module_url` can be replaced with a file path to a TSV file with your
+# desired gene list results
+gene_module_df <- readr::read_tsv(gene_module_url)
+```
+
+`read_tsv()` can read TSV files online and doesn't necessarily require you download the file first.
+Let's take a look at what this gene module list looks like.
```

Comment:
I didn't like the "look at... looks like" here. Also, I want to be clear about list vs. table, since they are different in R and I don't want people confused. (Don't @ me that a data frame is a special list)

```suggestion
Let's take a look at this gene module table.
```